### PR TITLE
[PW-3708] Cloning quote with quote helper on failed payment

### PR DESCRIPTION
**Pull request by acampos1916**
**Description**
When a payment for an order is refused a new order is ready to be placed using the same quote. In this PR the previous quote is being cloned (item lines, qtys, product options, coupon code, addresses) and replaced in the session so the following order is attached to it.

**Tested scenarios**
* CC
    * Non-3DS: failed payment does not place order -> successful payment uses the same quote and places the order
    * 3DS1: failed payment places `payment_review` order using quote ID and clones quote -> successful payment uses the cloned quote and places the order
    * 3DS2: failed payment places `canceled` order using quote ID and clones quote -> successful payment uses the cloned quote and places the order
* Redirect PMs
    * e.g. IDeal: failed payment places `new` order using quote ID and clones quote -> successful payment uses the cloned quote and places the order

**Fixed issue**:  PW-3708

### CODE REVIEW

**Review by msilvagarcia — DISMISSED** [review]

**Review by cyattilakiss — APPROVED** [review]

**Review by AlexandrosMor — APPROVED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Comment by acampos1916** [comment]
This is to be merged along with a feature toggle.

File: Controller/Process/Result.php [hunk 1]
```diff
@@ -69,9 +68,9 @@ class Result extends \Magento\Framework\App\Action\Action
     protected $quoteFactory;
```
Comment (peterojo) [inline Controller/Process/Result.php:68]: I think you can remove `$quoteFactory` as well

File: Helper/Quote.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Model\Quote as QuoteModel;
+use Magento\Quote\Model\Quote\AddressFactory as QuoteAddressFactory;
+use Magento\Quote\Model\ResourceModel\Quote\Address as QuoteAddressResource;
+use Magento\Sales\Model\Order;
+
+class Quote
+{
+    /**
+     * @var CartRepositoryInterface
```
Comment (peterojo) [inline Helper/Quote.php:40]:
```suggestion
     * @var QuoteRepository
```

File: Controller/Process/Result.php (hunk 1 above)
Comment (acampos1916) [reply Controller/Process/Result.php]: You're right!

**Review by peterojo — APPROVED** [review]

**Review by cyattilakiss — APPROVED** [review]

**Pull request merged**